src/screens/Dashboard/TripSearch.tsx

```diff
@@ -1,53 +1,535 @@
 import FullScreenHeader from '@atb/components/screen-header/full-header';
 import {RootStackParamList} from '@atb/navigation';
-import {StyleSheet, Theme} from '@atb/theme';
-import {useTranslation, TripSearchTexts} from '@atb/translations';
-import {CompositeNavigationProp} from '@react-navigation/native';
+import {StyleSheet, useTheme} from '@atb/theme';
+import {Language, TripSearchTexts, useTranslation} from '@atb/translations';
+import {
+  CompositeNavigationProp,
+  RouteProp,
+  useIsFocused,
+  useNavigation,
+} from '@react-navigation/native';
 import {StackNavigationProp} from '@react-navigation/stack';
-import React from 'react';
-import {View} from 'react-native';
-import {ScrollView} from 'react-native-gesture-handler';
-import {DashboardParams} from '../Dashboard';
-
-export type TripSearchScreenNavigationParams = StackNavigationProp<
-  DashboardParams,
-  'TripSearch'
->;
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import {
+  ActivityIndicator,
+  RefreshControl,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+import Button from '@atb/components/button';
+import {ExpandMore} from '@atb/assets/svg/mono-icons/navigation';
+import {ScrollView} from 'react-native';
+import {LocationInput, Section} from '@atb/components/sections';
+import {screenReaderPause} from '@atb/components/accessible-text';
+import {GeoLocation, Location, UserFavorites} from '@atb/favorites/types';
+import {Swap} from '@atb/assets/svg/mono-icons/actions';
+import {Location as LocationIcon} from '@atb/assets/svg/mono-icons/places';
+import ThemeIcon from '@atb/components/theme-icon';
+import {useFavorites} from '@atb/favorites';
+import {useLocationSearchValue} from '@atb/location-search';
+import {useGeolocationState} from '@atb/GeolocationContext';
+import {DashboardParams} from '.';
+import {SelectableLocationData} from '@atb/location-search/types';
+import Bugsnag from '@bugsnag/react-native';
+import {
+  getSearchTimeLabel,
+  SearchTime,
+  useSearchTimeValue,
+} from '../Assistant/journey-date-picker';
+import {TFunc} from '@leile/lobo-t';
+import {isInThePast} from '@atb/utils/date';
+import useTripsQuery from '../Assistant/use-trips-query';
+import {StaticColorByType} from '@atb/theme/colors';
+import ScreenReaderAnnouncement from '@atb/components/screen-reader-announcement';
+import Results from '../Assistant/Results';
```

Comment:
I would duplicate this from Assistant and have Dashboard _not_ be dependent on anything from Assistant. Keeping them separated and "deletable"

Reply:
ok